### PR TITLE
Add support for Elastic Network Interfaces

### PR DESCRIPTION
**Pull request by dhui**
I took a first pass of editing the README.

I only tested the `elastic_ip` and `elastic_network_interface` providers since those are the only providers that changed.
I tested the providers using the commands:

``` bash
KITCHEN_YAML=".kitchen.cloud.yml" AWS_KEYPAIR_NAME="test-kitchen" EC2_SSH_KEY_PATH="test-kitchen.pem" AWS_ACCESS_KEY_ID="XXXX" AWS_SECRET_ACCESS_KEY="XXXX" AWS_ELASTIC_IP="X.X.X.X" AWS_ELASTIC_NETWORK_INTERFACE_ID="eni-XXXX" AWS_ELASTIC_NETWORK_INTERFACE_DEVICE_INDEX="1" AWS_ELB_NAME="test-kitchen" AWS_S3_BUCKET="test-kitchen" AWS_S3_KEY="test-aws-cookbook-file" kitchen test elastic-ip
```

and

``` bash
KITCHEN_YAML=".kitchen.cloud.yml" AWS_KEYPAIR_NAME="test-kitchen" EC2_SSH_KEY_PATH="test-kitchen.pem" AWS_ACCESS_KEY_ID="XXXX" AWS_SECRET_ACCESS_KEY="XXXX" AWS_ELASTIC_IP="X.X.X.X" AWS_ELASTIC_NETWORK_INTERFACE_ID="eni-XXXX" AWS_ELASTIC_NETWORK_INTERFACE_DEVICE_INDEX="1" AWS_ELB_NAME="test-kitchen" AWS_S3_BUCKET="test-kitchen" AWS_S3_KEY="test-aws-cookbook-file" kitchen test elastic-network-interface
```

I modified, but didn't submit changes for, the `driver_config` section of `.kitchen.cloud.yml` to specify a `subnet_id` for testing the `elastic_network_interface` provider. I didn't submit the changes b/c not every test would needs the change and wasn't sure if you guys wanted to make `subnet_id` a required environment variable for test runs.

I also updated the public AMI images for all the specified platforms since I couldn't find the existing AMIs in the default region (`us-east-1`).

### CODE REVIEW

**Comment by dhui** [comment]
Bueller?

**Comment by iiro** [comment]
+1

File: README.md [hunk 1]
```diff
@@ -352,6 +373,31 @@ to reconnect to the instance with the new IP.
 You can also store this in a role as an attribute or assign to the
 node directly, if preferred.
 
+## aws_elastic_network_interface
+
+The `elastic_network_interface` resource provider does not support creating new
+network interfaces. This must be done before running a recipe that uses the resource.
+After creating a new Elastic Network Interface, we recommend storing it in a
+databag and loading the item in the recipe.
```
Comment (codeslinger) [inline README.md]: This doesn't reflect the reality of what the `:create` action actually does; needs updating.

File: README.md [hunk 2]
```diff
@@ -352,6 +373,31 @@ to reconnect to the instance with the new IP.
...
+To set up an existing Elastic Network Interface on a system:
+
+    eni_info = data_bag_item("aws", "eni")
+
+    aws_elastic_network_interface "eni" do
+      aws_access_key aws['aws_access_key_id']
+      aws_secret_access_key aws['aws_secret_access_key']
+      network_interface_id eni_info['network_interface_id']
+      device_index eni_info['device_index']
+      action [:create, :associate]
```
Comment (codeslinger) [inline README.md]: s/`:associate`/`:attach`/

**Comment by tas50** [comment]
Closing this out since we never heard back on the comments and master has diverged form this PR severely. If you want to update this to a custom resource against master I'd gladly take a look and get it merged in.

**Pull request closed without merging**